### PR TITLE
Common - Diagnose ace install if there are version problems

### PR DESCRIPTION
**Pull request by PabstMirror**

### CODE REVIEW

**Comment by LinkIsGrim** [comment]
Ref above: https://steamcommunity.com/sharedfiles/filedetails/?id=1744095754

**Comment by LinkIsGrim** [comment]
image

Logging works. I'd edit the "ace_component is outdated" message box to be something like this instead as well:
"ace_component is outdated (VERSION.NUMBER.NUMBER), common version is (COMMON.VERSION.NUMBER)"

**Comment by LinkIsGrim** [comment]
image
Thoughts?

**Comment by PabstMirror** [comment]
BTW this isn't meant to block or discourage custom builds or repacking/reuploading ace
that's still totally fine to do based on our license

it's just going to print more debug info on mod folders / workshop source and only when there is a problem

**Review by LinkIsGrim — APPROVED** [review]

**Review by jonpas — APPROVED** [review]

File: addons/common/functions/fnc_checkFiles.sqf [hunk 1]
```diff
@@ -39,29 +39,47 @@ if ([_cbaRequiredAr, _cbaVersionAr] call cba_versioning_fnc_version_compare) the
...
+        call FUNC(checkFiles_diagnoseACE);
 
         if ((_x select [0, 10]) != "ace_compat") then {
             if (hasInterface) then {
-                ["[ACE] ERROR", _errorMsg, {findDisplay 46 closeDisplay 0}] call FUNC(errorMessage);
+                _oldAddons pushBack _x;
             };
         } else {
-            _oldCompats pushBack _x;  // Don't block game if it's just an old compat pbo
+            _oldCompats pushBack [_x, _addonVersion];  // Don't block game if it's just an old compat pbo
```
Comment (PabstMirror) [inline addons/common/functions/fnc_checkFiles.sqf:58]:
```suggestion
            _oldCompats pushBack [_x, _addonVersion, _addonSource];  // Don't block game if it's just an old compat pbo
```
I think? not sure but looks like line 78 uses it?

Comment (LinkIsGrim) [reply addons/common/functions/fnc_checkFiles.sqf:58]: Forgot to remove after testing. Will just edit line 78.

**Pull request merged**